streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -210,11 +210,9 @@ public boolean isValidTransition(final State newState) {
     }
 
     private final Object stateLock = new Object();
-
     private volatile State state = State.CREATED;
-
     private KafkaStreams.StateListener stateListener = null;
-
+    private boolean inTheMiddleOfStateChange = false;
```

Comment:
`volatile`? It is potentially being accessed by different threads and the synchronization is on different objects